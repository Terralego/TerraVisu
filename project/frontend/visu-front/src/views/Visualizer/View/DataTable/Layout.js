import React from 'react';
import classnames from 'classnames';

export const Layout = ({ visible, children }) => (
  <div className={classnames({
    'data-table': true,
    'data-table--visible': visible,
  })}
  >
    {visible && children}
  </div>
);

export default Layout;
