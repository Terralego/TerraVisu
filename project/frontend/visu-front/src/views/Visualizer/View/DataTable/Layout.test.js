import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

it('should render correctly hidden', () => {
  const tree = renderer.create((
    <Layout>
      <p>Foo</p>
    </Layout>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly visible', () => {
  const tree = renderer.create((
    <Layout visible>
      <p>Foo</p>
    </Layout>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
