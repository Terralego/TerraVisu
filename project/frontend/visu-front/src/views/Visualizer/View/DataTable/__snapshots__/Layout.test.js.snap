// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly hidden 1`] = `
<div
  className="data-table"
/>
`;

exports[`should render correctly visible 1`] = `
<div
  className="data-table data-table--visible"
>
  <p>
    Foo
  </p>
</div>
`;
